microtiter_plates/libraries/microtiterplates/source.rb

```diff
@@ -231,6 +233,8 @@ def associate(index:, key:, data:)
   # @return [void]
   def associate_provenance(index:, key:, data:)
     to_item = @collection.part(index[0], index[1])
+    rescue 
+        raise index.class.to_s
```

Comment:
It's hard to tell what the boundaries of this rescue clause are. Can you follow the begin, rescue, end convention?

Reply:
Also, make the error message more informative, such as `"#{index.class.to_s} given, should be Array"`

Reply:
This rescue should have  been removed.  I don't think it is helpful in general.